doc/source/serve/doc_code/hpu_inference_serve.py

```diff
@@ -0,0 +1,154 @@
+            model_id_or_path,
+            config=hf_config,
+            torch_dtype=torch.float32,
+            low_cpu_mem_usage=True,
+            use_auth_token="",
+        )
+        model = model.eval().to(self.device)
+        # enable hpu graph runtime
+        from habana_frameworks.torch.hpu import (
+            wrap_in_hpu_graph,
+        )  # pylint: disable=E0401
+
+        self.model = wrap_in_hpu_graph(model)
+        # set pad token, etc.
+        self.model.generation_config.pad_token_id = 0
+        self.model.generation_config.bos_token_id = 1
+        self.model.generation_config.eos_token_id = 2
+        self.tokenizer.pad_token_id = self.model.generation_config.pad_token_id
+        self.tokenizer.padding_side = "left"
+
+        # async loop is used in streaming
+        self.loop = asyncio.get_running_loop()
+
+    # tokenize the input and pad according to length
+    def tokenize(self, prompt):
+        input_tokens = self.tokenizer(prompt, return_tensors="pt", padding=True)
+        return input_tokens.input_ids.to(device=self.device)
+
+    # generate the response given input
+    # return after it's all generated
+    def generate(self, prompt, **config):
+        input_ids = self.tokenize(prompt)
+        gen_tokens = self.model.generate(input_ids, **config)
+        return self.tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)[0]
+
+    # used in streaming
+    async def consume_streamer_async(self, streamer):
+        while True:
+            try:
+                for token in streamer:
+                    yield token
+                break
+            except Empty:
+                await asyncio.sleep(0.001)
+
+    # generate the repsonse given input in a streaming manner
+    def streaming_generate(self, prompt, streamer, **config):
+        input_ids = self.tokenize(prompt)
+        self.model.generate(input_ids, streamer=streamer, **config)
+
+    # handle requests
+    async def __call__(self, http_request):
+        json_request: str = await http_request.json()
+        prompts = []
+        text = json_request["text"]
+        config = json_request["config"] if "config" in json_request else {}
+        streaming_response = json_request["stream"]
+        # prepare prompts
+        if isinstance(text, list):
+            prompts.extend(text)
+        else:
+            prompts.append(text)
+        # process config
+        if "max_new_tokens" not in config:
+            # hpu requires setting max_new_tokens
+            config["max_new_tokens"] = 128
+        # enable hpu graph runtime
+        config["hpu_graphs"] = True
+        # lazy mode should be True when using hpu graphs
+        config["lazy_mode"] = True
+        # non streaming
+        if not streaming_response:
+            return self.generate(prompts, **config)
+        from starlette.responses import StreamingResponse
+        from transformers import TextIteratorStreamer
+        from functools import partial
```

Comment:
The `starlette.responses` and `functools` imports can be moved to the top level of the doc, outside the deployments.